src/main.rs

```diff
@@ -200,12 +200,13 @@ async fn application_trampoline(config: &Config) -> Result<()> {
 
                 // It's not even an encrypted file, so we need to keep the permission settings pretty tight.
                 // The only time I can really enforce that is when reading the password.
-                if metadata.mode() == 0o600 {
+                if metadata.mode() & 0o777 == 0o600 {
```

Comment:
On the Debian system I'm using, `metadata.mode()` returns `0o100600`, where `0o100000` indicates it's a regular file (see https://www.gnu.org/software/libc/manual/html_node/Testing-File-Type.html)